python_benchmarks/qr_codes/qr.py

```diff
@@ -33,19 +35,28 @@ def __init__(self):
         super().__init__()
         self.detector = cv.QRCodeDetector()
 
-    def detect(self, image):
-        ret, corners = self.detector.detectMulti(image)
-        if ret is False:
-            return False, np.array([])
-        self.detected_corners = corners
-        return ret, corners
-
-    def decode(self, image):
-        if self.detected_corners.size == 0:
-            return 0, [], None
-        r, decoded_info, straight_qrcode = self.detector.decodeMulti(image, self.detected_corners)
-        self.decoded_info = decoded_info
-        return r, decoded_info, straight_qrcode
+    def detect(self, image, use_alignment=True):
+            _, decoded_info, corners, _ = self.detector.detectAndDecodeMulti(image)
+            if corners is None or len(corners) == 0:
+                return False, np.array([])
+            self.decoded_info = decoded_info
+            self.detected_corners = corners
+            return True, corners
+
+    def detect_aruco(self, image):
+        self.detector.detectMultiAruco(image)
+
+    def decode(self, image, use_alignment=True):
+        if use_alignment:
+            if len(self.decoded_info) == 0:
+                return 0, [], None
```

Comment:
This part is the same for both `use_aligment` values and should be extracted.

Reply:
removed